components/NavigationBar/tests/unit/NavigationBarColorThemerTests.swift

```diff
@@ -38,4 +38,25 @@ class NavigationBarColorThemerTests: XCTestCase {
     XCTAssertEqual(navigationBar.titleTextColor, colorScheme.onPrimaryColor)
     XCTAssertEqual(navigationBar.tintColor, colorScheme.onPrimaryColor)
   }
+
+  func testSurfaceVariantColorThemerChangesTheCorrectParameters() {
+    // Given
+    let colorScheme = MDCSemanticColorScheme()
+    let navigationBar = MDCNavigationBar()
+    colorScheme.surfaceColor = .red
+    colorScheme.onSurfaceColor = .blue
+    navigationBar.backgroundColor = .white
+    navigationBar.titleTextColor = .white
+    navigationBar.tintColor = .white
+
+    // When
+    MDCNavigationBarColorThemer.applySemanticColorScheme(colorScheme,
+                                                         to: navigationBar,
+                                                         using: .surface)
+
+    // Then
+    XCTAssertEqual(navigationBar.backgroundColor, colorScheme.surfaceColor)
+    XCTAssertEqual(navigationBar.titleTextColor, colorScheme.onSurfaceColor)
```

Comment:
Please add a test for each of the enumeration values as well as one to verify that the "default" uses the `.primary` variant.